pyrefly/lib/test/dataclasses.rs

```diff
@@ -67,6 +67,234 @@ Data(0, 1)  # E: Argument `Literal[1]` is not assignable to parameter `y` with t
     "#,
 );
 
+testcase!(
+    test_replace,
+    r#"
+from dataclasses import dataclass, replace
+
+@dataclass
+class Foo:
+    x: int
+    y: str
+
+f = Foo(1, "a")
+
+replace(f, x="wrong")  # E: Argument `Literal['wrong']` is not assignable to parameter `x` with type `int` in function `dataclasses.replace`
+replace(f, z=3)  # E: Unexpected keyword argument `z` in function `dataclasses.replace`
+    "#,
+);
+
+testcase!(
+    test_replace_initvar_default,
+    r#"
+from dataclasses import dataclass, field, InitVar, replace
+
+@dataclass
+class WithInitVarDefault:
+    x: int
+    y: InitVar[str] = "ok"
+
+w = WithInitVarDefault(0)
+replace(w)
+replace(w, y="new")
+    "#,
+);
+
+testcase!(
+    test_replace_initvar_required,
+    r#"
+from dataclasses import dataclass, InitVar, replace
+
+@dataclass
+class Foo:
+    x: int
+    y: InitVar[int]
+
+f = Foo(1, 2)
+
+replace(f)  # E: Missing argument `y` in function `dataclasses.replace`
+    "#,
+);
+
+testcase!(
+    test_replace_positional_args_rejected,
+    r#"
+from dataclasses import dataclass, replace
+
+@dataclass
+class Foo:
+    x: int
+    y: str
+
+f = Foo(1, "a")
+
+replace(f, "extra")  # E: Expected 1 positional argument, got 2
+    "#,
+);
+
+testcase!(
+    test_replace_init_false_field_rejected,
+    r#"
+from dataclasses import dataclass, field, replace
+
+@dataclass
+class WithInitFalse:
+    x: int
+    y: int = field(init=False, default=5)
+
+g = WithInitFalse(1)
+
+replace(g, y=10)  # E: Unexpected keyword argument `y` in function `dataclasses.replace`
+    "#,
+);
+
+testcase!(
+    test_replace_classvar_rejected,
+    r#"
+from dataclasses import dataclass, replace
+from typing import ClassVar
+
+@dataclass
+class Config:
+    limit: int
+    MAX_ID: ClassVar[int] = 100
+
+c = Config(10)
+replace(c, limit=20)
+replace(c, MAX_ID=200) # E: Unexpected keyword argument `MAX_ID` in function `dataclasses.replace`
+    "#,
+);
+
+testcase!(
+    test_replace_union_mixed_dataclass,
+    r#"
+from dataclasses import dataclass, replace
+from typing import Union
+
+@dataclass
+class Foo:
+    x: int
+
+class Bar:
+    x: int
+
+def f(obj: Union[Foo, Bar]):
+    replace(obj, x=0)  # E: dataclasses.replace() expects a dataclass instance; got Bar | Foo
+    "#,
+);
+
+testcase!(
+    test_replace_union_two_dataclasses_rejects_bad_kw,
```

Comment:
Could you also add a test for an accepted `replace` call on a union of two dataclasses?